Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -10,13 +10,17 @@ extension MongoCollection {
      *   - options: Optional `FindOneAndDeleteOptions` to use when executing the command
      *
      * - Returns: The deleted document, represented as a `CollectionType`, or `nil` if no document was deleted.
+     *
      * - Throws: 
+     *   - `MongoError.invalidArgument` if any of the provided options are invalid
      *   - `MongoError.commandError` if there are any errors executing the command.
      *   - A `DecodingError` if the deleted document cannot be decoded to a `CollectionType` value
      */
     @discardableResult
-    public func findOneAndDelete(_ filter: Document, options: FindOneAndDeleteOptions? = nil) throws -> CollectionType? {
-        throw MongoError.commandError(message: "Unimplemented command")
+    public func findOneAndDelete(_ filter: Document,
+                                 options: FindOneAndDeleteOptions? = nil) throws -> CollectionType? {
+        let opts = options ?? FindOneAndDeleteOptions()
```

Comment:
the new tests I added led me to discover a subtle bug I've fixed here. basically, for `findOneAndDelete`, we *always* need to send options because we have to set the `remove` flag to true. I added a test that uses no options and it led to a crash. turns out since you have to always set `remove` or `update`, libmongoc does not account for the possibility of us passing in `nil` for the options. so I think I will refactor the logic in `findAndModify` to reflect that better